cpp/models/abm/world.h

```diff
@@ -42,33 +45,98 @@ namespace abm
  * @brief The World of the Simulation.
  * It consists of Location%s and Person%s (Agents).
  */
-class World
+class World : public CompartmentalModel<InfectionState, Populations<AgeGroup, InfectionState>, SimulationParameters>
 {
+    using Base = CompartmentalModel<InfectionState, mio::Populations<AgeGroup, InfectionState>, SimulationParameters>;
+
 public:
     using LocationIterator      = PointerDereferencingIterator<std::vector<std::unique_ptr<Location>>::iterator>;
     using ConstLocationIterator = PointerDereferencingIterator<std::vector<std::unique_ptr<Location>>::const_iterator>;
     using PersonIterator        = PointerDereferencingIterator<std::vector<std::unique_ptr<Person>>::iterator>;
     using ConstPersonIterator   = PointerDereferencingIterator<std::vector<std::unique_ptr<Person>>::const_iterator>;
+    /**
+     * @brief Create a World.
+     * @param[in] params Parameters of the simulation that are the same everywhere in the World.
+     */
+    World(SimulationParameters params)
+        : Base(Populations({params.get_num_groups(), InfectionState::Count}),
+               SimulationParameters(params.get_num_groups()))
+        , m_locations((uint32_t)LocationType::Count)
+        , m_infection_parameters(GlobalInfectionParameters(params.get_num_groups()))
+        , m_migration_parameters(MigrationParameters(params.get_num_groups()))
+        , m_simulation_parameters(params)
+        , m_trip_list()
+    {
+        use_migration_rules(true);
+    }
 
     /**
      * @brief Create a World.
-     * @param[in] params Parameters of the Infection that are the same everywhere in the World.
+     * @param[in] num_agegroups The number of age groups in the simulated world.
      */
-    World(const GlobalInfectionParameters& params = {})
-        : m_locations((uint32_t)LocationType::Count)
-        , m_infection_parameters(params)
-        , m_migration_parameters()
+    World(int32_t num_agegroups)
```

Comment:
In `add_persons`, there should be a check whether the AgeGroup passed is valid with respect to the set-up number of AgeGroups.

Reply:
A check is added as suggested.